Mathlib/Order/WithBot.lean

```diff
@@ -1122,104 +774,111 @@ theorem strictMono_map_iff {f : α → β} : StrictMono (WithTop.map f) ↔ Stri
   { WithTop.semilatticeSup, WithTop.semilatticeInf with }
 
-instance distribLattice [DistribLattice α] : DistribLattice (WithTop α) :=
-  { WithTop.lattice with
-    le_sup_inf := @le_sup_inf (WithBot αᵒᵈ)ᵒᵈ _ }
+instance distribLattice [DistribLattice α] : DistribLattice (WithTop α) where
+  le_sup_inf x y z := by
+    cases x <;> cases y <;> cases z <;> simp [← coe_inf, ← coe_sup]
+    simpa [← coe_inf, ← coe_sup] using le_sup_inf
 
 instance decidableEq [DecidableEq α] : DecidableEq (WithTop α) :=
   inferInstanceAs <| DecidableEq (Option α)
 
-instance decidableLE [LE α] [DecidableRel (α := α) (· ≤ ·)] :
-    DecidableRel (α := WithTop α) (· ≤ ·) := fun _ _ =>
-  decidable_of_decidable_of_iff toDual_le_toDual_iff
+instance decidableLE [LE α] [DecidableRel (α := α) (· ≤ ·)] : DecidableRel (α := WithTop α) (· ≤ ·)
+  | _, ⊤ => isTrue <| by simp
+  | ⊤, (a : α) => isFalse <| by simp
+  | (a : α), (b : α) => decidable_of_iff' _ coe_le_coe
 
-instance decidableLT [LT α] [DecidableRel (α := α) (· < ·)] :
-    DecidableRel (α := WithTop α) (· < ·) := fun _ _ =>
-  decidable_of_decidable_of_iff toDual_lt_toDual_iff
+instance decidableLT [LT α] [DecidableRel (α := α) (· < ·)] : DecidableRel (α := WithTop α) (· < ·)
+  | ⊤, _ => isFalse <| by simp
+  | (a : α), ⊤ => isTrue <| by simp
+  | (a : α), (b : α) => decidable_of_iff' _ coe_lt_coe
 
-instance isTotal_le [LE α] [IsTotal α (· ≤ ·)] : IsTotal (WithTop α) (· ≤ ·) :=
-  ⟨fun _ _ => by
-    simp_rw [← toDual_le_toDual_iff]
-    exact total_of _ _ _⟩
+instance isTotal_le [LE α] [IsTotal α (· ≤ ·)] : IsTotal (WithTop α) (· ≤ ·) where
+  total x y := by cases x <;> cases y <;> simp; simpa using IsTotal.total ..
 
-instance linearOrder [LinearOrder α] : LinearOrder (WithTop α) :=
-  Lattice.toLinearOrder _
+section LinearOrder
+variable [LinearOrder α] {x y : WithTop α}
 
-@[simp, norm_cast]
-theorem coe_min [LinearOrder α] (x y : α) : (↑(min x y) : WithTop α) = min (x : WithTop α) y :=
-  rfl
+instance linearOrder [LinearOrder α] : LinearOrder (WithTop α) := Lattice.toLinearOrder _
 
-@[simp, norm_cast]
-theorem coe_max [LinearOrder α] (x y : α) : (↑(max x y) : WithTop α) = max (x : WithTop α) y :=
-  rfl
+@[simp, norm_cast] lemma coe_min (a b : α) : ↑(min a b) = min (a : WithTop α) b := rfl
+@[simp, norm_cast] lemma coe_max (a b : α) : ↑(max a b) = max (a : WithTop α) b := rfl
 
-instance instWellFoundedLT [LT α] [WellFoundedLT α] : WellFoundedLT (WithTop α) :=
-  inferInstanceAs <| WellFoundedLT (WithBot αᵒᵈ)ᵒᵈ
+variable [DenselyOrdered α] [NoMaxOrder α]
+
+lemma le_of_forall_lt_iff_le : (∀ b : α, x < b → y ≤ b) ↔ y ≤ x := by
+  cases x <;> cases y <;> simp [exists_gt, forall_gt_imp_ge_iff_le_of_dense]
+
+lemma ge_of_forall_gt_iff_ge : (∀ a : α, a < x → a ≤ y) ↔ x ≤ y := by
+  cases x <;> cases y <;> simp [exists_gt, forall_lt_imp_le_iff_le_of_dense]
+
+end LinearOrder
+
+instance instWellFoundedLT [LT α] [WellFoundedLT α] : WellFoundedLT (WithTop α) where
```

Comment:
This proof is much larger than the previous one - I think the def-eq can stay.

If you still want to be proper about it, you could give `WithBot (αᵒᵈ) ≃o WithTop α` as an order isomorphism, and use `RelEmbedding.isWellFounded`.

Reply:
Reverted (for now)